x/exp/batch/batch_test.go

```diff
@@ -356,22 +358,77 @@ func TestBatchErrors(t *testing.T) {
 					types.NewEntityUID("Resource", "3"),
 				},
 			},
-		}, func(_ Result) {
+		}, func(_ Result) error {
 			total++
 			cancel()
+			return nil
 		},
 		)
 		testutil.ErrorIs(t, err, context.Canceled)
 		testutil.Equals(t, total, 1)
 	})
 
+	t.Run("lastContextCancelled", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		var total int
+		err := Authorize(ctx, cedar.NewPolicySet(), types.EntityMap{}, Request{
+			Principal: types.NewEntityUID("Principal", "principal"),
+			Action:    types.NewEntityUID("Action", "action"),
+			Resource:  Variable("resource"),
+			Context:   types.Record{},
+			Variables: Variables{
+				"resource": []types.Value{
+					types.NewEntityUID("Resource", "1"),
+					types.NewEntityUID("Resource", "2"),
+					types.NewEntityUID("Resource", "3"),
+				},
+			},
+		}, func(_ Result) error {
+			total++
+			if total == 3 {
+				cancel()
+			}
+			return nil
+		},
+		)
+		testutil.ErrorIs(t, err, context.Canceled)
+		testutil.Equals(t, total, 3)
+	})
+
+	t.Run("callbackErrored", func(t *testing.T) {
+		var total int
+		errWant := fmt.Errorf("errWant")
+		err := Authorize(context.Background(), cedar.NewPolicySet(), types.EntityMap{}, Request{
+			Principal: types.NewEntityUID("Principal", "principal"),
+			Action:    types.NewEntityUID("Action", "action"),
+			Resource:  Variable("resource"),
+			Context:   types.Record{},
+			Variables: Variables{
+				"resource": []types.Value{
+					types.NewEntityUID("Resource", "1"),
+					types.NewEntityUID("Resource", "2"),
+					types.NewEntityUID("Resource", "3"),
+				},
+			},
+		}, func(_ Result) error {
+			total++
+			if total == 2 {
+				return errWant
+			}
+			return nil
+		},
+		)
+		testutil.ErrorIs(t, err, errWant)
+		testutil.Equals(t, total, 2)
+	})
```

Comment:
Maybe add one more test case here that both returns an error and does a context cancel, and verifies that the returned error is both? As I think no tests here are currently exercising the behaviour that the callback error is joined with the context error.